packages/next/build/webpack-config.ts

```diff
@@ -424,6 +424,27 @@ export default async function getBaseWebpackConfig(
     )
   }
 
+  const getPackagePath = (name: string, contextDirectory: string) => {
+    const packageJsonPath = require.resolve(`${name}/package.json`, {
+      paths: [contextDirectory],
+    })
+    return path.dirname(packageJsonPath)
+  }
+
+  const nextPackagePath = getPackagePath('next', dir)
+  const reactPackagePath = getPackagePath('react', dir)
+
+  // Packages which will be split into the 'framework' chunk.
+  // Only top-level packages are included, e.g. nested copies like
+  // 'node_modules/react-scroll-parallax/node_modules/prop-types' are not included.
+  const topLevelFrameworkPaths = [
```

Comment:
This should be a Set so that the lookup isn't slow

Reply:
Using a Set wouldn't help in this case, because the `topLevelFrameworkPaths.some()` expression below does a linear search anyway.

We're checking whether the `module.resource` string starts with one of the paths in `topLevelFrameworkPaths`, so it can't be simplified to a constant-time lookup.

Reply:
I don't think it was doing that when I made the comment but I can't remember and you've force pushed so I don't know.